app/src/main/kotlin/com/intive/picover/profile/view/ProfileScreen.kt

```diff
@@ -89,11 +92,18 @@ fun ProfileScreen(viewModel: ProfileViewModel) {
 			.padding(top = 10.dp),
 		horizontalAlignment = Alignment.CenterHorizontally,
 	) {
-		if (profile.isLoaded()) {
-			ProfileSegment(profile.data(), photoLauncher, sheetState, coroutineScope)
-		} else {
-			// TODO NAN change to linear indicator for first loading and after refresh to avoid jumping UI
-			CircularProgressIndicator()
+		when (profile) {
+			is Loaded -> {
+				ProfileSegment(profile.data(), photoLauncher, sheetState, coroutineScope)
+			}
+
+			is Loading -> {
+				CircularProgressIndicator()
+			}
+
+			is Error -> {
+				ErrorProfileSegment(viewModel::fetchProfile)
```

Comment:
@intive/android-guild-szczecin 
Or, if we would like I can go with our generic error placeholder:
<p align="center">
 <img src="https://github.com/intive/Picover/assets/57491794/d8ddb602-d532-4068-a965-4442f4611d78" width="350">
</p>

Reply:
Ok, I will go with a generic solution. What about the loader I have a special view for this in another PR because IMO it is a specific situation with the loader.